This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:

Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear

From b5d790047628344eb1a3ea1e85935dc50fdaa444 Mon Sep 17 00:00:00 2001
From: Will Huang <wilhuang@qti.qualcomm.com>
Date: Mon, 15 Dec 2025 14:43:41 +0530
Subject: [PATCH] WDS EXT: Set sta aid for WDS EXT sta id

For MCC association is not handle in hostapd, so sta->aid is not
updated, but WDS EXT feature need it as a sta index to distinguish
different agent station, so set it by calling hostapd_get_aid().

aid clear logic already exists in ap_free_sta().

Change-Id: Ib998298176a9420c91f5ea8326576edd7dd49ab9
---
 src/ap/ieee802_11.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 8327d3b0f8..979123c000 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -6924,6 +6924,13 @@ void ieee802_11_rx_from_unknown(struct hostapd_data *hapd, const u8 *src,
 			int ret;
 			char ifname_wds[IFNAMSIZ + 1];
 
+			if (hostapd_get_aid(hapd, sta) < 0) {
+				wpa_printf(MSG_DEBUG,
+					  "No room for aid " MACSTR,
+					  MAC2STR(src));
+				return;
+			}
+
 			wpa_printf(MSG_DEBUG, "Enable 4-address WDS mode for "
 				   "STA " MACSTR " (aid %u)",
 				   MAC2STR(sta->addr), sta->aid);
-- 
2.34.1

